### PR TITLE
MergeTree: Fix split handling for local references

### PR DESCRIPTION
**Pull request by anthony-murphy**
In this change I extended the insertAtRefPos workload we used in conflict farm to randomly introduce transient, simple, and slide on remove local references.

- transient references are not added to the segment or its local reference collection
- simple are added to the segment, but will not survive segment delete
- slide on removed will survive delete, and slide to the nearest positions

In adding this workload i hit an assert in the split handling, so i fixed that bug, and also added a regression case for that specific issue. 

related to #1008

### CODE REVIEW

File: packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts [hunk 1]
```diff
@@ -44,9 +44,13 @@ export const insertAtRefPos: TestOperation =
         if(segs.length > 0) {
             const text = client.longClientId!.repeat(random.integer(1, 3)(mt));
             const seg = random.pick(mt,segs);
-            return client.insertAtReferencePositionLocal(
-                new LocalReference(client, seg, random.integer(0, seg.cachedLength - 1)(mt)),
-                TextSegment.make(text));
+            const lref = new LocalReference(
+                client, seg, random.integer(0, seg.cachedLength - 1)(mt),
+                random.pick(mt,[ReferenceType.Simple, ReferenceType.SlideOnRemove, ReferenceType.Transient]));
+            if(lref.refType !== ReferenceType.Transient) {
```
Comment (Abe27342) [inline packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts:50]: What makes Transient references special here? (in general docs on all of the ReferenceTypes would be great :))

**Review by Abe27342 — APPROVED** [review]
Looks good to me aside from a couple formatting nits. I'm going to make a separate PR adding some doc to LocalReference/LocalReferenceCollection that I think would have been helpful in understanding what was going on here.

File: packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts:50]: This is all legacy code with no documentation, so i don't have full clarity. They are also reused between a number of features, markers, local references, and shared intervals. Some types make some sense for some scenarios, but not others. It's an area that needs quite a bit of clean up.

Comment (anthony-murphy) [reply packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts:50]: Within merge tree itself, slide on remove has special handling. when the segment is removed only local references with that type survive.

**Pull request merged**